test/macro.jl

```diff
@@ -9,34 +9,57 @@ Symbolics.@register_symbolic fff(t)
 @test isequal(fff(t), Symbolics.Num(Symbolics.Term{Real}(fff, [Symbolics.value(t)])))
 
 const SymMatrix{T,N} =  Symmetric{T, AbstractArray{T, N}}
+many_vars = @variables t=0 a=1 x[1:4]=2 y(t)[1:4]=3 w[1:4] = 1:4 z(t)[1:4] = 2:5 p(..)[1:4]
+
+let
+    @register_array_symbolic ggg(x::AbstractVector) begin
+        container_type=SymMatrix
+        size=(length(x) * 2, length(x) * 2)
+        eltype=eltype(x)
+    end false
+
+    ## @variables
+
+    gg = ggg(x)
+
+    @test ndims(gg) == 2
+    @test size(gg) == (8,8)
+    @test eltype(gg) == Real
+    @test symtype(unwrap(gg)) == SymMatrix{Real, 2}
```

Comment:
Why is this `SymMatrix{Real, 2}` and the other `SymMatrix{Real}`?

Reply:
We can't infer `ndims` if it's not specified in the macrocall, since `promote_symtype` only gets the types of arguments to the function